sound/soc/sof/ipc4-control.c

```diff
@@ -66,7 +66,7 @@ static int sof_ipc4_set_get_kcontrol_data(struct snd_sof_control *scontrol,
 		 * configuration
 		 */
 		memcpy(scontrol->ipc_control_data, scontrol->old_ipc_control_data,
-		       scontrol->max_size);
+		       scontrol->size);
```

Comment:
Ok this took a while to parse, but I think this is correct. So without this patch, incorrect amount was copied as:
 - too little copied as scontrol->max_size doesn't cover the internal header struct
 - too much copied as scontrol->size may be smaller than max_size